test/test_util.h

```diff
@@ -581,6 +584,14 @@ __host__ __device__ __forceinline__ void InitValue(GenMode gen_mode, T &value, i
         }
         break;
     }
+#else
+     case RANDOM:
+     case RANDOM_BIT:
+     case RANDOM_MINUS_PLUS_ZERO:
+         _CubLog("%s\n",
+                 "cub::InitValue cannot generate random numbers on device.");
+         cub::ThreadTrap();
```

Comment:
This breaks NVC++.  A bunch of the NVHPC tests have been failing since this was checked in.

This change uses the NVCC split compilation model.  It doesn't work with NVC++'s unified compilation model.  NVC++ only sees the device version of the random options, and will use that for both host and device.  Not only will it not run correctly, it won't even compile, because it contains a call to the device-only function `cub::ThreadTrap` which causes failures during host code generation.

This change needs to use a host/device mechanism that works with both NVCC and NVC++.  Either `<nv/target>` or the older (and much more cumbersome) `if (CUB_IS_HOST_CODE)`.

Reply:
d'oh, I should have thought of this. I'll rewrite this to use the `CUB_IS_HOST_CODE` pattern for now since the `<nv/target>` refactor is still a WIP.